apps/fz_http/test/support/api_case.ex

```diff
@@ -0,0 +1,61 @@
+defmodule FzHttpWeb.ApiCase do
+  @moduledoc """
+  This module defines the test case to be used by
+  tests that require setting up a connection.
+
+  Such tests rely on `Phoenix.ConnTest` and also
+  import other functionality to make it easier
+  to build common data structures and query the data layer.
+
+  Finally, if the test case interacts with the database,
+  we enable the SQL sandbox, so changes done to the database
+  are reverted at the end of every test. If you are using
+  PostgreSQL, you can even run database tests asynchronously
+  by setting `use FzHttpWeb.ConnCase, async: true`, although
+  this option is not recommended for other databases.
+  """
+  use ExUnit.CaseTemplate
+  use FzHttp.CaseTemplate
+
+  alias FzHttp.{
+    ApiTokensFixtures,
+    UsersFixtures
+  }
+
+  using do
+    quote do
+      # Import conveniences for testing with connections
+      import Plug.Conn
+      import Phoenix.ConnTest
+      import FzHttp.TestHelpers
+
+      use FzHttpWeb, :verified_routes
+
+      # The default endpoint for testing
+      @endpoint FzHttpWeb.Endpoint
+    end
+  end
+
+  def new_conn do
+    Phoenix.ConnTest.build_conn()
+  end
+
+  def api_conn do
+    new_conn()
+    |> Plug.Conn.put_req_header("accept", "application/json")
+  end
+
+  def authed_conn do
+    user = UsersFixtures.user(%{role: :admin})
+    api_token = ApiTokensFixtures.api_token(%{"user_id" => user.id})
+    {:ok, token, _claims} = FzHttpWeb.Auth.JSON.Authentication.fz_encode_and_sign(api_token, user)
+
+    api_conn()
+    |> Plug.Conn.put_req_header("authorization", "bearer #{token}")
+    |> FzHttpWeb.Auth.JSON.Pipeline.call([])
+  end
+
+  setup _tags do
+    {:ok, unauthed_conn: api_conn(), authed_conn: authed_conn()}
```

Comment:
We should not put any default tags here, instead, tests can either use their own setup block (so that you can see if the block needs auth or not on `describe` level)

Reply:
Noted. Updated, though there's a bit more code duplication now than I'd prefer. Open to cleaning it up a tad or leaving it for explicit reasons